lib/auth/auth_test.go

```diff
@@ -2126,56 +2126,134 @@ func TestDeleteMFADeviceSync(t *testing.T) {
 	err = authServer.SetAuthPreference(ctx, authPreference)
 	require.NoError(t, err)
 
-	clt, err := testServer.NewClient(TestUser(username))
+	userClient, err := testServer.NewClient(TestUser(username))
 	require.NoError(t, err)
 
-	// Insert dummy devices.
-	webDev1, err := RegisterTestDevice(ctx, clt, "web-1", proto.DeviceType_DEVICE_TYPE_WEBAUTHN, nil /* authenticator */)
-	require.NoError(t, err)
-	_, err = RegisterTestDevice(ctx, clt, "web-2", proto.DeviceType_DEVICE_TYPE_WEBAUTHN, webDev1)
-	require.NoError(t, err)
-	totpDev1, err := RegisterTestDevice(ctx, clt, "otp-1", proto.DeviceType_DEVICE_TYPE_TOTP, webDev1, WithTestDeviceClock(testServer.Clock()))
-	require.NoError(t, err)
-	_, err = RegisterTestDevice(ctx, clt, "otp-2", proto.DeviceType_DEVICE_TYPE_TOTP, webDev1, WithTestDeviceClock(testServer.Clock()))
-	require.NoError(t, err)
+	// webDev1 is used as the authenticator for various checks.
+	webDev1, err := RegisterTestDevice(ctx, userClient, "web1", proto.DeviceType_DEVICE_TYPE_WEBAUTHN, nil /* authenticator */)
+	require.NoError(t, err, "RegisterTestDevice(web1)")
+
+	// Insert a few devices.
+	// Devices with the "delete-" prefix are deleted by following tests.
+	totpOpts := []TestDeviceOpt{WithTestDeviceClock(testServer.Clock())}
+	var allDevs []*TestDevice
+	for _, spec := range []struct {
+		name       string
+		deviceType proto.DeviceType
+		opts       []TestDeviceOpt
+	}{
+		{
+			name:       "delete-web1",
+			deviceType: proto.DeviceType_DEVICE_TYPE_WEBAUTHN,
+		},
+		{
+			name:       "delete-web2",
+			deviceType: proto.DeviceType_DEVICE_TYPE_WEBAUTHN,
+		},
+		{
+			name:       "delete-totp1",
+			deviceType: proto.DeviceType_DEVICE_TYPE_TOTP,
+			opts:       totpOpts,
+		},
+		{
+			name:       "delete-totp2",
+			deviceType: proto.DeviceType_DEVICE_TYPE_TOTP,
+			opts:       totpOpts,
+		},
+		{
+			name:       "spare-web1",
```

Comment:
A comment explaining why "spares" are necessary, to see `TestDeleteMFADeviceSync_lastDevice` for more, could be clarifying.

Reply:
I've changed the second_factor mode to "optional" and dropped the spares, that should make it slightly simpler.